pennylane/templates/tensornetworks/mps.py

```diff
@@ -155,42 +193,60 @@ def __init__(
                     f"Weights tensor must have last dimension of length {n_params_block}; got {shape[-1]}"
                 )
 
-        self._hyperparameters = {"ind_gates": ind_gates, "block": block}
-        super().__init__(template_weights, wires=wires, id=id)
+        self._weights = template_weights
+        self._hyperparameters = {"ind_gates": ind_gates, "block": block, **kwargs}
+
+        if self._weights is None:
+            super().__init__(wires=wires, id=id)
+        else:
+            super().__init__(self._weights, wires=wires, id=id)
+
+    @property
+    def num_params(self):
+        """int: Number of trainable parameters that the operator depends on."""
+        return 0 if self._weights is None else 1
```

Comment:
If `num_params=1` does that mean that we only have one *argument* that is trainable?

Reply:
Yes.